### PR TITLE
[flutter_tools] remove globals from IntelliJ validator, refactor tests to remove dependency on JAR

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Move the intellij validator to its own file, and split off the tests. Remove globals from each, and remove dependency on a real jar/filesystem by creating a minimal version of the manifest xml and setting it up in the memory file system.

https://github.com/flutter/flutter/issues/47161

### CODE REVIEW

File: packages/flutter_tools/lib/src/intellij/intellij_validator.dart [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import 'package:meta/meta.dart';
+
+import '../base/file_system.dart';
+import '../base/platform.dart';
+import '../base/user_messages.dart' hide userMessages;
+import '../base/version.dart';
+import '../doctor.dart';
+import '../ios/plist_parser.dart';
+import 'intellij.dart';
+
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/intellij/intellij_validator.dart:14]: This logic is unchanged exception for globals

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Google testing failed, working on g3fix

**Comment by jonahwilliams** [comment]
Google testing passed

**Pull request merged**